pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -136,10 +137,25 @@ func NewDefaultClientConfigLoadingRules() *ClientConfigLoadingRules {
 
 	envVarFiles := os.Getenv(RecommendedConfigPathEnvVar)
 	if len(envVarFiles) != 0 {
-		chain = append(chain, filepath.SplitList(envVarFiles)...)
```

Comment:
Seems like the chain should be allowed to contain directories and you handle the directories when processing the chain.  That way you'll be able to integrate with the existing error handling scheme.